lib/class-wp-rest-widgets-controller.php

```diff
@@ -116,7 +92,7 @@ public function register_routes() {
 			),
 			'schema' => array( $this, 'get_public_item_schema' ),
 		) );
-		register_rest_route( $this->namespace, '/' . $this->rest_base .'/types/(?P<type>[\w-]+)', array(
+		register_rest_route( $this->namespace, '/' . $this->rest_base_types . '/(?P<type>[\w-]+)', array(
```

Comment:
@NateWr I wrote a widgets endpoint using be/v1/widgets/:id and I think /wp/v2/widgets/:type is probably better moving forward.  The id of a widget is only necessary information on get item, update item, and delete item requests.  For any request of the collection, or any request to create a new widget /:id just ends up being clunky.  By separating the pages-2 into pages/2 it just makes requests match better to what actually needs to be done in the request, in my opinion. I think a structure like this would make most sense but I am open to anything.

GET /wp/v2/widgets/ (get_items, collection from widget factory.)
GET /wp/v2/widgets/:type (get_items, collection from widget factory for specific type.)
GET /wp/v2/widgets/:type/:number (get_item, specific widget from widget factory.)

The widgets-types endpoint I don't believe is necessary and should be tacked on as schema, via OPTIONS request to /wp/v2/widgets/, because wp/v2/widget-types/ only returns schema.
We could also do an OPTIONS request for /wp/v2/widgets/:type same effect except schema data just for that type of widget.

Since widget-types is currently just spitting out schema info, I think it makes sense to bake that into the actual endpoints.

Now onto the last three of CRUD (create, update, delete). Lets talk creation first. I suggest:

Create:
POST wp/v2/widgets/:type

I think it would be nice to do it this way but wp/v2/widgets/?widget_type=calendar works fine as well.  Basically it would make the request and endpoint more synonymous and query args could be left to be sidebar_id, for the sidebar we are placing the widget in, and sidebar_position for the position the widget should fall into the sidebar.

Update:
POST wp/v2/widgets/:type/:number

This will have query arguments that match up with the schema for this widget.

Delete:
DELETE wp/v2/widgets/:type/:number

In order for this to work we will need to separate the second route into two routes.
Instead of 'wp/v2/widgets' . '/(?<id_base>[\w-]+/<number>[\d]+)' (Forgive me if the syntax is wrong, just the idea counts.)

We would register another route with just the /wp/v2/widgets/(?<id_base>[\w-]+).  This will enable us to have our create_item in there.  It is not terribly necessary, but I believe it would make the structuring of the endpoints rather clean if we did this.